responsemanager/queryexecutor.go

```diff
@@ -129,12 +129,14 @@ func (qe *queryExecutor) executeQuery(
 				code = graphsync.RequestFailedUnknown
 				return nil
 			}
-			if err == errCancelledByCommand {
+			if err == runtraversal.ErrFirstBlockLoad {
+				code = graphsync.RequestFailedContentNotFound
+			} else if err == errCancelledByCommand {
```

Comment:
It looks like in some of the error cases handled here we set the code then return nil.
But in these new error cases we don't return after setting the code. Just want to confirm that's correct?

Reply:
Yea so, basically the above two cases involve no final network message to send (see `rb.ClearRequest`)

While these cases do involve a final message back to the requestor. (see `rb.FinishWithError`). Ultimately they return nil two lines below that. This is a lot of nested conditionals and I agree it's confusing and merits a refactor, but yes, it's correct :)

Reply:
Cool just wanted to make sure 👍